ksqldb-engine/src/main/java/io/confluent/ksql/engine/SandboxedExecutionContext.java

```diff
@@ -160,7 +161,11 @@ public ExecuteResult execute(
     ).execute(ksqlPlan.getPlan());
 
     // Having a streams running in a sandboxed environment is not necessary
-    result.getQuery().map(QueryMetadata::getKafkaStreams).ifPresent(streams -> streams.close());
+    if (getKsqlConfig().getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)) {
+      result.getQuery().map(QueryMetadata::getKafkaStreams).ifPresent(streams -> streams.close());
+    } else {
+      engineContext.getQueryRegistry().getPersistentQueries().values().stream().map(QueryMetadata::getKafkaStreams).forEach(KafkaStreams::close);
```

Comment:
This is going to close the actual runtime -- note that when we create the `SandboxedBinPackedPersistentQueryMetadataImpl` it saves a reference to the runtime of the original `BinPackedPersistentQueryMetadataImpl`. This is one of the things we need to fix

Reply:
It seems like we should only ever be creating SandboxedSharedKafkaStreamsRuntimeImpl in one place, specifically we should create them in the `QueryRegistryImpl` constructor like we do now. But we then need to connect the `SandboxedBinPackedPersistentQueryMetadataImpl` to the sandboxed runtime rather than let it have a reference to the actual runtime, to avoid it closing that

Reply:
But I think this definitely explains why the validation runtimes aren't being closed

Reply:
That seems more like an issue with the sandbox version of the query registry

Reply:
but I have a work around